pkg/operator/ceph/cluster/osd/spec.go

```diff
@@ -467,6 +467,7 @@ func (c *Cluster) makeDeployment(osdProps osdProperties, osd OSDInfo, provisionC
 	if osdProps.onPVC() {
 		k8sutil.AddLabelToDeployment(OSDOverPVCLabelKey, osdProps.pvc.ClaimName, deployment)
 		k8sutil.AddLabelToPod(OSDOverPVCLabelKey, osdProps.pvc.ClaimName, &deployment.Spec.Template)
+		k8sutil.AddLabelToPod(OSDDeviceClassSetLabelKey, osdProps.name, &deployment.Spec.Template)
```

Comment:
Let's add it both to the pod and the deployment, just like `OSDOverPVCLabelKey` has both.

Reply:
Done.